app/components/index.js

```diff
@@ -15,22 +16,46 @@ export default class Index extends Component {
 
   constructor(props) {
     super(props);
-    this.state = { isLoading: true, sliderRecord: [], postRecord: [] }
+    this.state = { 
+      isLoading: true, 
+      no_more: false,
+      moreIsLoading: false,
+      sliderRecord: [], 
+      postRecord: [],
+      page:0
+   }
   }
 
-  componentDidMount() {
-    return fetch('https://9ff6ba98.ngrok.io/api/v0/home_posts?next_page=1')
+  makeRemoteRequest = (chinese) => {
+   
+    let next_page = this.state.page + 1;
+    if (this.state.no_more || this.state.moreIsLoading) return null;
+    if (next_page > 1) this.setState({moreIsLoading: true});
+    let lang = chinese ? 'cn' : 'en';
+    
+    let request_uri = `${BASE_REQUEST_URI}next_page=${next_page}&l=${lang}`;
+    console.log(this.state.page);
+    return fetch(request_uri)
       .then((response) => response.json())
```

Comment:
you need to handle response with error status correctly https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
```
fetch(url).then(function(response) {
  if(response.ok) {
    return response.json();
  }
  throw new Error('Network response was not ok.');
}).then(function(responseJson) { 
  /* do stuff here */
}).catch(function(error) {
  console.log('There has been a problem with your fetch operation: ' + error.message);
});
```